src/solver/tron.jl

```diff
@@ -0,0 +1,397 @@
+# Some parts of this code were adapted from
+# https://github.com/PythonOptimizers/NLP.py/blob/develop/nlp/optimize/tron.py
+
+using LinearOperators, NLPModels
+
+export tron
+
+function active(x, l, u; rtol::Real = 1e-8, atol::Real = 1e-8)
+  A = Int[]
+  n = length(x)
+  for i = 1:n
+    if l[i] == x[i] == u[i]
+      push!(A, i)
+    elseif x[i] < l[i] + min(rtol * (u[i]-l[i]), atol)
+      push!(A, i)
+    elseif x[i] > u[i] -  min(rtol * (u[i]-l[i]), atol)
+      push!(A, i)
+    end
+  end
+  return A
+end
```

Comment:
If we drop Julia 0.4 support, is this a good use case of 0.5's new broadcast feature?